router/pkg/watcher/simple_watcher.go

```diff
@@ -0,0 +1,64 @@
+package watcher
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type SimpleWatcherOptions struct {
+	Interval time.Duration
+	Logger   *zap.Logger
+	Path     string
+	Callback func()
+}
+
+func LogSimpleWatch(ctx context.Context, options SimpleWatcherOptions) {
```

Comment:
as we basically require all options to be set, would it make sense to make them private and have a function `NewSimpleWatcherOptions()` with a bit of validation that returns an error instead of a panicking when calling Logger or Callback?

Reply:
Maybe, I could also split it into like `NewSimpleWatcher()` and `func (*SimpleWatcher) Start(ctx)` or something.

Reply:
It has error handling and a sort of "init" phase now, with minimal bloat to the API